source/scripting/Game.cs

```diff
@@ -338,137 +390,253 @@ public static InputMode CurrentInputMode
+		/// </summary>
+		/// <param name="index">The Input Method (0 = Mouse and Keyboard, 2 = GamePad)</param>
+		/// <param name="control">The <see cref="Control"/></param>
+		/// <param name="value">the value to set the control to</param>
 		public static void SetControlNormal(int index, Control control, float value)
 		{
 			Function.Call(Hash._SET_CONTROL_NORMAL, index, control, value);
 		}
 
+		/// <summary>
+		/// Pauses or Resumes the game
+		/// </summary>
+		/// <param name="value">if set to <c>true</c> Pause the game; otherwise, resume the game</param>
 		public static void Pause(bool value)
 		{
 			Function.Call(Hash.SET_GAME_PAUSED, value);
 		}
+		/// <summary>
+		/// Pauses or Resumes the game clock
+		/// </summary>
+		/// <param name="value">if set to <c>true</c> Pause the game clock; otherwise, resume the game clock</param>
 		public static void PauseClock(bool value)
 		{
 			Function.Call(Hash.PAUSE_CLOCK, value);
 		}
 
+		/// <summary>
+		/// Performs and automative game save
+		/// </summary>
 		public static void DoAutoSave()
 		{
 			Function.Call(Hash.DO_AUTO_SAVE);
 		}
+		/// <summary>
+		/// Shows the save menu enabling the user to perform a manual game save.
+		/// </summary>
 		public static void ShowSaveMenu()
 		{
 			Function.Call(Hash.SET_SAVE_MENU_ACTIVE, true);
 		}
 
-		public static void FadeScreenIn(int time)
-		{
-			Function.Call(Hash.DO_SCREEN_FADE_IN, time);
-		}
-		public static void FadeScreenOut(int time)
-		{
-			Function.Call(Hash.DO_SCREEN_FADE_OUT, time);
-		}
-
+		/// <summary>
+		/// Returns a localised <see cref="string"/> from the games language files with a specified GXT key
+		/// </summary>
+		/// <param name="entry">The GXT key</param>
+		/// <returns>The localised <see cref="string"/> if the key exists; otherwise, <see cref="string.Empty"/></returns>
 		public static string GetGXTEntry(string entry)
 		{
-			return Function.Call<string>(Hash._GET_LABEL_TEXT, entry);
+			if (Function.Call<bool>(Hash.DOES_TEXT_LABEL_EXIST, entry))
+			{
+				return Function.Call<string>(Hash._GET_LABEL_TEXT, entry);
+			}
+			return "";
 		}
 
+		/// <summary>
+		/// Calculates a Jenkins One At A Time hash from the given <see cref="string"/> which can then be used by any native function that takes a hash
+		/// </summary>
+		/// <param name="input">The input <see cref="string"/> to hash</param>
+		/// <returns>The Jenkins hash of the <see cref="string"/></returns>
 		public static int GenerateHash(string input)
 		{
 			if (string.IsNullOrEmpty(input))
 			{
 				return 0;
 			}
 
-			uint hash = 0;
-			char[] chars = input.ToCharArray();
-
-			//converts ascii uppercase to lowercase
-			for (int i = 0, length = chars.Length; i < length; i++)
-			{
-				if (chars[i] >= 'A' && chars[i] <= 'Z')
-				{
-					chars[i] = (char)(chars[i] + 32);
-				}
-			}
-
-			byte[] bytes = Encoding.UTF8.GetBytes(chars);
+			uint hash = 0;	  
+			
+			byte[] bytes = Encoding.ASCII.GetBytes(input.ToLower());
```

Comment:
Just looking over it again, it actually uses both. from what i can see if the string starts with a quote it ignores the quote, then uses ASCII uppercase to lowercase, then hashes until end of string or another quote is found.
If string doesnt start with a quote then it uses UTF8 combined with ASCII uppercase to lowercase on the whole string.